lib/client/Client.js

```diff
@@ -638,14 +680,22 @@ export default class Client {
   }
 
   _handleMailboxesSetResponse(response) {
-    return MailboxesSet.fromJSONObject(this, response[1]);
```

Comment:
As per the spec, `response[0]` is the name of the response, like messagesSet, or vacationResponseSet, etc. `response[1]` is the actual response data.